src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -110,14 +111,14 @@
   private Handler<Void> closeHandler;
 
   // storage of PUBLISH QoS=1 messages which was not responded with PUBACK
-  private LinkedHashMap<Integer, io.netty.handler.codec.mqtt.MqttMessage> qos1outbound = new LinkedHashMap<>();
+  private HashSet<Integer> qos1outbound = new HashSet<>();
 
   // storage of PUBLISH QoS=2 messages which was not responded with PUBREC
   // and PUBREL messages which was not responded with PUBCOMP
-  private LinkedHashMap<Integer, io.netty.handler.codec.mqtt.MqttMessage> qos2outbound = new LinkedHashMap<>();
+  private HashSet<Integer> qos2outbound = new HashSet<>();
```

Comment:
the message was there because there is another PR related to restoring messages from storage when the client crashes and the inflight queue has to be rebuilt, so the client is able to resend messages to the server.

Reply:
ok, we can keep the map then right ?

Reply:
Yes. The PR should be just about avoiding the memory leak removing the message from the `qos2inbound`

Reply:
ok, I'll keep though the usage of HashMap over LinkedHashMap that does not seem to be useful